### PR TITLE
Disposable batteries, magazine wells, and other battery fixes!

### PR DESCRIPTION
**Pull request by Inglonias**
#### Summary
```SUMMARY: Content "Added disposable batteries and fine-tuned battery magazines"```

#### Purpose of change
Partially addresses #30213, among other things

#### Describe the solution
* Most battery powered tools now have valid magazine wells.
* Disposable batteries have been added. They have twice the capacity of the highest capacity of a given size, but cannot be recharged. They can spawn in the same places as other batteries of their size.
* Made a sanity check on all tools. I noticed that I missed the electric firestarter, rendering it useless. It now takes magazines like everything else. I also noticed, for instance, that the electric combatsaw took heavy batteries and the electric chainsaw took medium ones, which was wrong. They now both take medium batteries.

#### Describe alternatives you've considered
N/A

#### Additional context
There's a lot here that I've been meaning to do for a while now. The next step is to figure out why cell phones are acting strangely and to properly resolve professions that start with a cell phone so I can remove the strange hack I currently use.

### CODE REVIEW

**Comment by esotericist** [comment]
Thanks again for continuing to work on this.

It's been reported, though, that there's a bit of a display error when it comes to battery cells. In particular, "high capacity" batteries have a name that is too long to show their current charge in the Advanced Inventory Manager.

image

Is there something we can do about the nomenclature here to make that a little less bulky?

**Comment by Inglonias** [comment]
Shortening names is definitely something I can do. Thanks for bringing that to my attention.

**Comment by Inglonias** [comment]
I've shortened the names of the high capacity batteries by removing the word "cell" from the name. Hopefully that's enough to at least help out.

The word "battery" has now lost all meaning to me for the next few hours. Battery.

File: data/json/items/ammo.json [hunk 1]
```diff
@@ -199,7 +263,8 @@
     "id": "heavy_atomic_battery_cell",
     "type": "MAGAZINE",
     "category": "spare_parts",
-    "name": "atomic heavy battery cell",
+    "name": "atomic heavy battery",
+    "name": "atomic heavy batteries",
```
Comment (kevingranade) [inline data/json/items/ammo.json]: whoopsie

**Comment by Inglonias** [comment]
Merge conflicts resolved.

**Pull request merged**